FirebasePerformance/Sources/FPRNanoPbUtils.m

```diff
@@ -0,0 +1,501 @@
+    reachabilityRef = SCNetworkReachabilityCreateWithName(kCFAllocatorSystemDefault, "google.com");
+  });
+
+  SCNetworkReachabilityFlags reachabilityFlags = 0;
+  SCNetworkReachabilityGetFlags(reachabilityRef, &reachabilityFlags);
+
+  // Parse the network flags to set the network type.
+  if (reachabilityFlags & kSCNetworkReachabilityFlagsReachable) {
+    if (reachabilityFlags & kSCNetworkReachabilityFlagsIsWWAN) {
+      networkType = firebase_perf_v1_NetworkConnectionInfo_NetworkType_MOBILE;
+    } else {
+      networkType = firebase_perf_v1_NetworkConnectionInfo_NetworkType_WIFI;
+    }
+  }
+
+  return networkType;
+}
+
+#ifdef TARGET_HAS_MOBILE_CONNECTIVITY
+/** Get the current cellular network connection type in
+ * firebase_perf_v1_NetworkConnectionInfo_MobileSubtype format.
+ *  @return Current cellular network connection type.
+ */
+static firebase_perf_v1_NetworkConnectionInfo_MobileSubtype FPRCellularNetworkType() {
+  static NSDictionary<NSString *, NSNumber *> *cellularNetworkToMobileSubtype;
+  static dispatch_once_t onceToken = 0;
+  dispatch_once(&onceToken, ^{
+    cellularNetworkToMobileSubtype = @{
+      CTRadioAccessTechnologyGPRS : @(firebase_perf_v1_NetworkConnectionInfo_MobileSubtype_GPRS),
+      CTRadioAccessTechnologyEdge : @(firebase_perf_v1_NetworkConnectionInfo_MobileSubtype_EDGE),
+      CTRadioAccessTechnologyWCDMA : @(firebase_perf_v1_NetworkConnectionInfo_MobileSubtype_CDMA),
+      CTRadioAccessTechnologyHSDPA : @(firebase_perf_v1_NetworkConnectionInfo_MobileSubtype_HSDPA),
+      CTRadioAccessTechnologyHSUPA : @(firebase_perf_v1_NetworkConnectionInfo_MobileSubtype_HSUPA),
+      CTRadioAccessTechnologyCDMA1x : @(firebase_perf_v1_NetworkConnectionInfo_MobileSubtype_CDMA),
+      CTRadioAccessTechnologyCDMAEVDORev0 :
+          @(firebase_perf_v1_NetworkConnectionInfo_MobileSubtype_EVDO_0),
+      CTRadioAccessTechnologyCDMAEVDORevA :
+          @(firebase_perf_v1_NetworkConnectionInfo_MobileSubtype_EVDO_A),
+      CTRadioAccessTechnologyCDMAEVDORevB :
+          @(firebase_perf_v1_NetworkConnectionInfo_MobileSubtype_EVDO_B),
+      CTRadioAccessTechnologyeHRPD : @(firebase_perf_v1_NetworkConnectionInfo_MobileSubtype_EHRPD),
+      CTRadioAccessTechnologyLTE : @(firebase_perf_v1_NetworkConnectionInfo_MobileSubtype_LTE)
+    };
+  });
+
+  NSString *networkString = NetworkInfo().currentRadioAccessTechnology;
+  NSNumber *cellularNetworkType = cellularNetworkToMobileSubtype[networkString];
+  return cellularNetworkType.intValue;
+}
+#endif
+
+/** Reorders the list of sessions to make sure the first session is verbose if at least one session
+ *  in the list is verbose.
+ *  @return Ordered list of sessions.
+ */
+NSArray<FPRSessionDetails *> *MakeFirstSessionVerbose(NSArray<FPRSessionDetails *> *sessions) {
+  NSMutableArray<FPRSessionDetails *> *orderedSessions =
+      [[NSMutableArray<FPRSessionDetails *> alloc] initWithArray:sessions];
+
+  __block NSInteger firstVerboseSessionIndex = -1;
```

Comment:
Why not a plain for loop to iterate through the array? The block overhead seems like expensive overhead for just checking one field of each session.

On further research, it might be just a style difference - https://stackoverflow.com/questions/4486622/when-to-use-enumerateobjectsusingblock-vs-for

Reply:
Thanks for the reference! I still changed it to a plain for loop to make it more straightforward.